synonymdev/bitkit-react-native
# feat: show backup status

# Description

There are 2 types of backup in Bitkit. And we need to validate them differently

#### Backpack ####

Data just pushed to the seeder using backpack client and SlashtagsRPC. Because SlashtagsRPC promise doens't throw timeout error, we just memorise time of the last attempt. If it was too long ago (30 minutes), then we show an error and failed state in Setting/Backup screen.

#### Profile and Contacts Hypercores #####

They are synced in the background so we still can't get backup status easily. So to do that we periodically ask Seeder-server if it has these hypercores. If it does - backup is successfull. 

One more difference here, that Backpack backups can be easily retried, by just altering redux state (`backup.remoteLdkActivityBackupSynced` for example). But we can't do that for Hypercores, so refresh button is disabled for them.

---

## Type of change
New feature

## Tests
Unit test

## Screen shot / Video

![image](https://user-images.githubusercontent.com/155891/235461092-b99f18af-3b0b-4824-9847-25db1bf6b3f1.png)


## QA Notes

- Turn device into airplate mode
- Change settings or add a few tags
- wait 30-40 minutes and then you should see backup failed warning
- go to setting/back up screen and check backup status

## Bitkit Version
